<launch>
  <node pkg="waypoint_control"
    type="waypoint_control"
    name="waypoint_control_node"
    output="screen">
    <!-- DO NOT CHANGE NAME FROM waypoint_control_node, this package is coded to look there -->

    <!-- Safety params -->
    <param name="vx_max" value="1.5"/>
    <param name="vy_max" value="1.5"/>
    <param name="vz_max" value="1.5"/>
    <param name="ax_max" value="0.85"/>
    <param name="ay_max" value="0.85"/>
    <param name="az_max" value="1.2"/>
    <param name="vmax_for_timing" value="3"/> <!--Used for coarse timing/takeoff/etc -->
    <param name="vmax_real" value="4"/> <!-- Used to determine need for intermediate waypoints -->
    <param name="waypointHitDist" value="0.15"/> <!-- Must be this close to record passage through wpt -->
        <!-- Set wptHitDist=0 to enable timing option -->
    
    <!-- Position Control parameters -->
    <param name="mass" value="0.7762"/>       <!-- kg -->
    <param name="gz" value="9.81"/>          <!-- m/s^2 -->
    <param name="thrustRatio" value="1.5"/>  <!-- maxThrust/Weight -->
    <param name="kpX" value="3.4"/>
    <param name="kpY" value="3.4"/>
    <param name="kpZ" value="2.2"/>
    <param name="kvX" value="2.9"/>
    <param name="kvY" value="2.9"/>
    <param name="kvZ" value="1.9"/>
    <param name="kiX" value="0.25"/>
    <param name="kiY" value="0.25"/>
    <param name="kiZ" value="0.3"/>
    <param name="maxInteg_X" value="1.5"/>
    <param name="maxInteg_Y" value="1.5"/>
    <param name="maxInteg_Z" value="1.5"/>

    <!-- Arena parameters -->
    <param name="xCenter" value="0.0"/> <!-- dist btwn origin of arena center and takeoff location -->
    <param name="yCenter" value="0.0"/>
    <param name="zCenter" value="0.0"/>
    <!-- NOTE: gps_kf sets initial pose/vel to 0 -->

    <!-- Misc parameters -->
    <param name="quadName" type="string" value="Valkyrie"/>
    <param name="quadPoseTopic" type="string" value="Valkyrie/local_odom"/>
    <param name="quadWaypointTopic" type="string" value="Valkyrie/refPose1"/> <!-- topic for single wpts -->
    <param name="quadWaypointListTopic" type="string" value="PosControl/PVATrajectory/Quad1"/> <!-- topic for lists of wpts -->
    <param name="publishPVA_Topic" type="string" value="px4_control/PVA_Ref_test"/>
    <param name="gps_fps" value="20"/>

  </node>
</launch>